api/konnect/v1alpha1/konnect_controlplane_types.go

```diff
@@ -0,0 +1,97 @@
+package v1alpha1
+
+import (
+	sdkkonnectgocomp "github.com/Kong/sdk-konnect-go/models/components"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func init() {
+	SchemeBuilder.Register(&KonnectControlPlane{}, &KonnectControlPlaneList{})
+}
+
+// KonnectControlPlane is the Schema for the KonnectControlplanes API.
+//
+// +genclient
+// +kubebuilder:resource:scope=Namespaced
+// +kubebuilder:object:root=true
+// +kubebuilder:object:generate=true
+// +kubebuilder:subresource:status
+// +kubebuilder:printcolumn:name="Programmed",description="The Resource is Programmed on Konnect",type=string,JSONPath=`.status.conditions[?(@.type=='Programmed')].status`
+// +kubebuilder:printcolumn:name="ID",description="Konnect ID",type=string,JSONPath=`.status.id`
+// +kubebuilder:printcolumn:name="OrgID",description="Konnect Organization ID this resource belongs to.",type=string,JSONPath=`.status.organizationID`
+// +kubebuilder:validation:XValidation:rule="!has(oldSelf.spec.konnect.authRef) || has(oldSelf.spec.konnect.authRef)", message="Konnect Configuration's API auth ref reference is immutable"
```

Comment:
Should the XValidation rule be like `!has(oldSelf.spec.konnect.authRef) || (oldSelf.spec.konnect.authRef == self.pec.konnect.authRef)`? The rule now seems always true.

Reply:
This was indeed broken. b6eedde0762afcb18e3081ded32d776a00a60603 fixes that.

This rule was based on https://kubernetes.io/blog/2022/09/29/enforce-immutability-using-cel/#immutablility-after-first-modification but we don't really need that because authRef is required (and it's `name` field).